pkg/console/console.go

```diff
@@ -0,0 +1,171 @@
+package console
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Builder provides a struct for console object from the cluster and a console definition.
+type Builder struct {
+	// Console definition, used to create the pod object.
+	Definition *v1.Console
+	// Created console object.
+	Object *v1.Console
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewBuilder creates a new instance of Builder.
+func NewBuilder(apiClient *clients.Settings, name string) *Builder {
+	if name == "" {
```

Comment:
Code inconsistency comparing to other packages. 
Can you move this check after builder init. Put it on line 42.
Please use our regular handler:

```
	if name == "" {
		glog.V(100).Infof("The name of the Console is empty")

		builder.errorMsg = "MachineConfig 'name' cannot be empty"
	}
```

Reply:
I don't understand why you do not accept different from the error reaction on the empty field when only one single and same for all deployment option is available. I'm still insist we do not need name parameter at all in this case.